contracts/test/Verifiers.ts

```diff
@@ -0,0 +1,181 @@
+    });
+
+    it("should verify grand sum proof", async () => {
+      // Concatenates the snark proof and the grand sum proof
+      let snarkProofArray = ethers.utils.arrayify(commitmentCalldata.range_check_snark_proof);
+      let grandSumProofArray = ethers.utils.arrayify(commitmentCalldata.grand_sums_batch_proof);
+      let totalBalances = commitmentCalldata.total_balances;
+
+      // The first 64 bytes of the snark proof represent a commitment to the corresponding to the user identity
+      // Starting from the next 64 bytes, each set of 64 bytes represents commitments corresponding to the total sum of balances
+      let grandSumCommitments = snarkProofArray.slice(64, (64 + grandSumProofArray.length));
+
+      // The verifier iterates over points in the proofs while verifying them.
+      // The proofs look like:
+      //  i = 0                                       1                                               N                 
+      // [grand_sum_proof_p1_x, grand_sum_proof_p1_y, grand_sum_proof_p2_x, grand_sum_proof_p2_y, ... grand_sum_proof_pN_x, grand_sum_proof_pN_y, ...]
+      // [    snark_proof_p1_x,     snark_proof_p1_y,     snark_proof_p2_x,     snark_proof_p2_y, ...     snark_proof_pN_x,     snark_proof_pN_y, ...] 
+      //  Where `N` is the number of currencies
+      let proofs = ethers.utils.concat([grandSumProofArray, grandSumCommitments]);
+
+      expect(await grandSumVerifier.verifyProof(verifyingKey.address, proofs, totalBalances)).to.be.not.reverted;
+    });
+  });
+
+
+  describe("Inclusion Proof Verifier", () => {
+    let inclusionVerifier: InclusionVerifier;
+    let verifyingKey: Halo2VerifyingKey;
+    let inclusionProof: BytesLike;
+    let commitmentCalldata: {
+      range_check_snark_proof: BytesLike;
+      grand_sums_batch_proof: BytesLike;
+      total_balances: BigNumber[];
+    };
+    let challenges: [BigNumber, BigNumber, BigNumber, BigNumber];
+    let userId: BytesLike;
+    let userIdBigUint: BigNumber;
+    let balance1: BigNumber;
+    let balance2: BigNumber;
+
+    beforeEach(async () => {
+      const deploymentInfo = await loadFixture(deployVerifyingFixture);
+      verifyingKey = deploymentInfo.verifyingKey;
+      commitmentCalldata = deploymentInfo.commitmentCalldata;
+
+      inclusionVerifier = await ethers.deployContract(
+        "src/InclusionVerifier.sol:InclusionVerifier"
+      ) as InclusionVerifier;
+      await inclusionVerifier.deployed();
+
+      verifyingKey = deploymentInfo.verifyingKey;
+
+      const inclusionJson = fs.readFileSync(
+        path.resolve(
+          __dirname,
+          "../../kzg_prover/bin/inclusion_proof_solidity_calldata.json"
+        ),
+        "utf-8"
+      );
+      const inclusionCalldata: any = JSON.parse(inclusionJson);
+
+      inclusionProof = inclusionCalldata.proof;
+      userId = inclusionCalldata.user_id;
+      challenges = inclusionCalldata.challenges;
+      userIdBigUint = inclusionCalldata.user_values[0];
```

Comment:
Do we still need the `userId` if we have it in `inclusionCalldata.user_values[0]`?

Reply:
Actually 'userId' is not used anywhere in the test code. I will remove it on the test code.
Would it be better to remove it from the csv file as well?

Reply:
I don't think we can remove it from the csv, it is used in some tests (`e.g., test_valid_univariate_grand_sum_full_prover`) and should be used in production as well.